### ISSUE TITLE
ty incorrectly thinks all objects have an `__mro__` attribute

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#986]
### Summary

```py
reveal_type(print.__mro__)  # revealed: tuple[<class 'FunctionType'>, <class 'object'>]
reveal_type(object().__mro__)  # revealed: tuple[<class 'object'>]
```

^Both those attribute accesses fail at runtime, but ty doesn't emit any diagnostics here.
